src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,533 @@
+    # - mondo_label_lookup: For looking up Mondo labels
+    # - source_data_map: keys=source ids, values(mondo_id (if any), source_id, mondo_label (if any))
+    # - mondo_data_map: keys=mondo ids, values=(mondo_id, source_id (if any), mondo_label)
+    # - mondo_ids: List of set of Mondo IDs from mondo.sssom.tsv
+    # - source_ids: List of set of source IDs from mondo.sssom.tsv
+    print('- Collecting all source terms and mondo mappings')
+    # - Mappings
+    ss_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    #   - label lookups
+    mondo_label_lookup: Dict[CURIE, str] = ss_df.set_index('subject_id')['subject_label'].to_dict()
+    #   - filter: Only exact matches
+    ss_df = ss_df[ss_df['predicate_id'] == 'skos:exactMatch']
+    # todo: ordo fix: temp until fixed: https://github.com/monarch-initiative/mondo-ingest/issues/369
+    if onto_config_path.endswith('ordo.yml'):
+        ss_df['object_id'] = ss_df['object_id'].apply(lambda _id: _id.replace('orphanet.ordo', 'Orphanet'))
+    #   - filter: Only in source
+    ss_df['obj_prefix'] = ss_df['object_id'].apply(lambda _id: _id.split(':')[0])
+    _col_renames = {'subject_id': 'mondo_id', 'subject_label': 'mondo_label', 'object_id': 'source_id'}
+    ss_df_source_dicts: List[Dict] = ss_df[ss_df['obj_prefix'].isin(owned_prefix_map)][
+        ['subject_id', 'subject_label', 'object_id']].rename(columns=_col_renames).to_dict(orient='records')
+    # - Lookup dictionaries
+    #   - mapping lookups
+    mondo_source_map: Dict[CURIE, List[CURIE]] = {}
+    for d in ss_df_source_dicts:
+        mondo_source_map.setdefault(d['mondo_id'], []).append(d['source_id'])
+    source_mondo_map: Dict[CURIE, CURIE] = {d['source_id']: d['mondo_id'] for d in ss_df_source_dicts}
+    #   - metadata lookups
+    mondo_data_map: Dict[CURIE, Dict] = _get_mondo_data_map(ss_df_source_dicts, mondo_label_lookup)
+    source_data_map: Dict[CURIE, Dict] = {d['source_id']: d for d in ss_df_source_dicts}
+    #   - ID set lists
+    mondo_ids: List[CURIE] = list(mondo_data_map.keys())
+    source_ids: List[CURIE] = list(source_data_map.keys())
+
+    # Get subclass edges: ancestors (direct + indirect)  ---------------------------------------------------------------
+    # - ancestors_mondo_mondo: Ancestors (Mondo IDs) from Mondo (direct and indirect)
+    # - ancestors_mondo_source: Ancestors (source IDs) from Mondo (direct and indirect)
+    # - ancestors_mondo_mondo_and_1or2_ids_not_in_source: Edges from ancestors_mondo_mondo not in source (unused)
+    # - ancestors_source_source: Ancestors (source IDs) from source (direct and indirect)
+    # - ancestors_source_mondo: Ancestors (Mondo IDs) from source (direct and indirect)
+    # - ancestors_source_mondo_and_1or2_ids_not_in_mondo: Edges from ancestors_source_mondo not in Mondo (unused)
+    print('- Collecting ancestors (direct + indirect')
+    ancestors_mondo_mondo: Set[RELATIONSHIP] = _ancestors(  # from Mondo
```

Comment:
## Inefficiency: Mondo ancestors queried for every source @matentzn 
@matentzn Choose what to do (not hold up for this PR though).

<details><summary>Details</summary>
<p>

This impacts performance but doesn't need to hold up this PR.

On my very good computer it takes ~6 minutes to query all ancestor relations in Mondo.

Logically, this only needs to be done 1 time whenever we run the "synchronization (subclass)" pipeline. However, this is being done for each source. There's 6 sources, so that's 5 excessive times, meaning the pipeline will currently take an extra 30 minutes longer.

## Possible solutions

a. Do nothing
b. Query Mondo ancestors as prerequisite before running for each source
  - File type
    a. Pickle (this is how I'm currently doing it
    b. TSV
  - Commit this output? Probably not; put it in `.gitignore` (e.g. in `tmp/`)

</p>
</details>

Reply:
6 min is ok for now.

Reply:
Well it's not really 6 minutes, it's 6 * 5 = 30 minutes.

Reply:
I really dont like to make such calls. Ok. 30 min is terrible but building a mega cache is worse. Lets leave this issue for a future refactoring.

Reply:
Ok. I have a third solution with another tradeoff you might like.
<details><summary>Details</summary>
<p>

_Added this alternative solution to my comment above:_

c. Fix by using new `make` pattern
Instead of running the same script for every source, we run the script for all sources at once, and it queries Mondo 1x at beginning.

</p>
</details>

Reply:
approved